### PR TITLE
feat(assets): Update text to mention we support images in MDX and Markdoc too

### PR DESCRIPTION
**Pull request by Princesseuh**
#### What kind of changes does this PR include?

- New or updated content

#### Description

We added support for relative and optimized images in MDX and Markdoc as well! Core PR for MDX: https://github.com/withastro/astro/pull/6824 (Markdoc is already in, but figure we can do both at once.)

### CODE REVIEW

File: src/content/docs/en/guides/assets.mdx [hunk 1]
```diff
@@ -77,11 +77,11 @@ import rocket from '../assets/images/rocket.svg'
 <img src={rocket.src} width={rocket.width} height={rocket.height} alt="A rocketship in space." />
 ```
 
-### Update your Markdown files
+### Update your Markdown, MDX and Markdoc files
```
Comment (delucis) [inline src/content/docs/en/guides/assets.mdx:80]: Here before @Yan-Thomas 😤 

This list needs an Oxford comma:

```suggestion
### Update your Markdown, MDX, and Markdoc files
```

File: src/content/docs/en/guides/assets.mdx [hunk 2]
```diff
@@ -77,11 +77,11 @@ import rocket from '../assets/images/rocket.svg'
...
 
-Relative images can now be referenced in Markdown files. These will automatically be optimized and hashed by Astro's build process.
+Relative images can now be referenced in Markdown, MDX and Markdoc files. These will automatically be optimized and hashed by Astro's build process.
```
Comment (delucis) [inline src/content/docs/en/guides/assets.mdx:82]: Ditto:

```suggestion
Relative images can now be referenced in Markdown, MDX, and Markdoc files. These will automatically be optimized and hashed by Astro's build process.
```

File: src/content/docs/en/guides/assets.mdx [hunk 3]
```diff
@@ -10,7 +10,7 @@ import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
 The new assets experience currently features:
 
 - A new built-in `<Image />` component
-- Relative images with automatic optimization in Markdown
+- Relative images with automatic optimization in Markdown, [MDX](/en/guides/integrations-guide/mdx/) and [Markdoc](/en/guides/integrations-guide/markdoc/)
```
Comment (delucis) [inline src/content/docs/en/guides/assets.mdx:13]:
```suggestion
- Relative images with automatic optimization in Markdown, [MDX](/en/guides/integrations-guide/mdx/), and [Markdoc](/en/guides/integrations-guide/markdoc/)
```

File: src/content/docs/en/guides/assets.mdx [hunk 4]
```diff
@@ -77,11 +77,11 @@ import rocket from '../assets/images/rocket.svg'
...
 ```
 
-### Update your Markdown files
+### Update your Markdown, MDX and Markdoc files
 
-Relative images can now be referenced in Markdown files. These will automatically be optimized and hashed by Astro's build process.
+Relative images can now be referenced in Markdown, MDX and Markdoc files. These will automatically be optimized and hashed by Astro's build process.
 
-This allows you to move your images from the `public/` directory to the new, reserved `src/assets/` directory or move them near your Markdown files. (See the URL path of these built images in the example below.)
+This allows you to move your images from the `public/` directory to the new, reserved `src/assets/` directory or move them near your Markdown, MDX or Markdoc files. (See the URL path of these built images in the example below.)
```
Comment (delucis) [inline src/content/docs/en/guides/assets.mdx:84]:
```suggestion
This allows you to move your images from the `public/` directory to the new, reserved `src/assets/` directory or move them near your Markdown, MDX, or Markdoc files. (See the URL path of these built images in the example below.)
```

**Review by delucis — COMMENTED** [review]
Looks good to me! Just the same comma nit four times over.

**Review by sarah11918 — APPROVED** [review]
LGTM!

**Review by delucis — APPROVED** [review]

**Review by Yan-Thomas — APPROVED** [review]

**Comment by sarah11918** [comment]
@Princesseuh It looks like this feature was already merged? So merge this one whenever you're ready!  :partying_face:

**Pull request merged**